### ISSUE TITLE
EDA report fails if target is numerical or integer variable

### ISSUE THREAD

**Issue by neural-oracle** [choonghyunryu/dlookr#75]
Hi, what a great package!

Would be great, if the EDA report works also for a target variable, that is numeric or integer.

Using the documentation example the following command throws an error (i just switched the target from the binary death_event to the numerical age variable! (same for the "cpk_enzyme" as target or any other numerical variable of the same heartfailure-df.

heartfailure %>%
  eda_web_report(target = "age", subtitle = "heartfailure", 
                 output_dir = "./", output_file = "EDA.html", theme = "blue")

error: 
Quitting from lines 361-373 (eda_temp.Rmd) 
 Error in UseMethod("select") : 
no applicable method for 'select' applied to an object of class "c('relate', 'lm')"

**Comment by neural-oracle** [comment]
great ... thx a loot for your kind response... 

I also tried to convert the numeric target to cut into 5 classes and convert it to an ordered factor ... 
Even with an ordered factor it did not work for me ... 

It works for a binary target (as you mentioned) ... so maybe i did sth wrong ... but you can check yourself if it works with a categorical variable with more than two classes ... thx .. and really great, great work!

**Comment by neural-oracle** [comment]
thx Choon ... for your kind answer...
I just wanted to let you know ... I cut the numeric variable into 5 bins ... and converted it to an ordered factor ... the EDA_web_report function works with the ordered factor ... as promised by the documentation ... so this is completely sufficient for me ... since it dlookr is not made for numeric variables as I understand from your answer .. .i think this is not a bug (as you flagged .. but rather an enhancement ..) ... I just thought .. to let you know this ... Cheers

**Comment by neural-oracle** [comment]
and one last final remark :)
This works:
  heartfailure %>%  
    eda_web_report(target = "age_bin", subtitle = "heartfailure", 
                 output_dir = "./", output_file = "EDA.html", theme = "blue")

This does not work (ceteris paribus - only function changed from eda_web_report to eda_pdf_report and ending from .html to .pdf):
  heartfailure %>%  
    eda_paged_report(target = "age_bin", subtitle = "heartfailure", 
                 output_dir = "./", output_file = "EDA.pdf", theme = "blue")

error for eda_paged_report:
Quitting from lines 250-251 (eda_paged_temp.Rmd) 
Error in `select()`:
! Can't subset columns that don't exist.
✖ Column `variable` doesn't exist.
Backtrace:
  1. dlookr:::html_paged_target_numerical(...)
  5. dplyr:::select.data.frame(., -variable)
  8. tidyselect::eval_select(expr(c(...)), .data)
  9. tidyselect:::eval_select_impl(...)
 18. tidyselect:::vars_select_eval(...)
     ...
 21. tidyselect:::reduce_sels(node, data_mask, context_mask, init = init)
 22. tidyselect:::walk_data_tree(new, data_mask, context_mask)
 23. tidyselect:::as_indices_sel_impl(...)
 24. tidyselect:::as_indices_impl(x, vars, call = call, strict = strict)
 25. tidyselect:::chr_as_locations(x, vars, call = call)
Warning messages:
1: In for (j in seq_along(x)) { :
  closing unused connection 3 (.//eda_paged_temp.Rmd)
2: 'dlookr::plot_correlate' ist veraltet.
Benutzen Sie stattdessen 'plot.correlate'
Siehe help("Deprecated")

**Comment by choonghyunryu** [comment]
I have two answers.

- The first answer is that eda_web_report() supports that the target variable is numeric. I misunderstood and made the answer wrong. So, I will find and fix the error in the issue you raised in the issue report.

- The second answer is,
    - The following error has been fixed in dlookr version 0.6.0. Try upgrading.

```
error for eda_paged_report:
Quitting from lines 250-251 (eda_paged_temp.Rmd)
Error in select():
! Can't subset columns that don't exist.
✖ Column variable doesn't exist.
```

**Comment by choonghyunryu** [comment]
eda_web_report() was fixed. I will also modify eda_paged_report().

To use the modified eda_web_report(), install version 0.6.1 of the development Github.

**Comment by choonghyunryu** [comment]
Now, the development version of github supports numeric target variables in eda_web_report() and eda_paged_report().

**Comment by wolass** [comment]
Hi! I just downloaded the package from github a couple minutes ago ( "dlookr 0.6.1.9000 2022-07-10)

The code: 

```
heartfailure %>%
eda_web_report(target = "age", subtitle = "heartfailure",
output_dir = "./", output_file = "EDA.html", theme = "blue")
```

still produces the same error 

```
Quitting from lines 361-373 (eda_temp.Rmd) 
Error in UseMethod("select") : 
  no applicable method for 'select' applied to an object of class "c('relate', 'lm')"

```

**Comment by neural-oracle** [comment]
> I want to thank you too, for the immediate response and bug fix!
I downloaded dlookr 0.6.1.9 and tested it with another dataset. My target ist also a positive integer variable (like age in heartfailure). My regressor are percentile transformed variable on the scale [0,1]. In order to exclude a problem with 0 and na, I shifted the scale by one to [1,2] and used the na.omit(df) %>% ... command.
I tested several sets of features (all on the scale [1,2]) ... In some cases it worked (the eda_web_report) ... but in some cases I ran into following error:

Quitting from lines 361-373 (eda_temp.Rmd) 
Error in seq_len(n) : argument must be coercible to non-negative integer

I made sure that there are no NAs and that there are no negative or 0 values in my dataset. I just wanted to let you know about this problem. Here are the last few lines of the traceback error log:
Error in seq_len(n) : argument must be coercible to non-negative integer
42.
stop(e)
41.
value[[3L]](cond)
40.
tryCatchOne(expr, names, parentenv, handlers[[1L]])
39.
tryCatchList(expr, classes, parentenv, handlers)
38.
tryCatch({
result <- withVisible(rlang::eval_tidy(expr))
if (result$visible) {
capture.output(print(result$value)) ...
37.
capturePlot(~{
plot(fit_lm)
}, device = device, width = width * pixelratio, height = height * 
pixelratio, res = 72 * pixelratio)
36.
rlang::eval_tidy(rlang::expr(capturePlot({
{
expr
} ...
35.
htmltools::plotTag({
plot(fit_lm)
}, "Visuzlization of relation", width = 600, height = 400, device = grDevices::png)
34.
(function (index) 
{
nm_var <- nm_numeric[index]
fit_lm <- relate(tgt_by, all_of(nm_var)) ...
33.
do.call(func, args[seq_len(numArgs)])
32.
callFunc(details, index, key)
31.
FUN(X[[i]], ...)
30.
lapply(seq_len(nrow(data)), function(index) {
callFunc(details, index, key)
})
29.
FUN(X[[i]], ...)
28.
lapply(columnKeys, function(key) {
column <- list(accessor = key, name = key, type = colType(data[[key]]))
if (!is.null(columns[[key]])) {
column <- mergeLists(column, columns[[key]]) ...
27.
reactable(., defaultColDef = colDef(style = "font-size: 14px;color: hsl(0, 0%, 40%);"), 
columns = list(target_variable = colDef(name = "Target Variable")), 
details = function(index) {
nm_var <- nm_numeric[index] ...
26.
tab_main %>% reactable(defaultColDef = colDef(style = "font-size: 14px;color: hsl(0, 0%, 40%);"), 
columns = list(target_variable = colDef(name = "Target Variable")), 
details = function(index) {
nm_var <- nm_numeric[index] ...
25.
html_target_numerical(reportData, target = targetVariable, base_family = base_family) at <text>#2
24.
eval(expr, envir, enclos)

Thx again for this great package!!

**Comment by choonghyunryu** [comment]
@neural-oracle,
Please share your reproducible code.

**Comment by choonghyunryu** [comment]
@neural-oracle,

I have found the cause of this problem.
This happened while using the plot.relate() function from the dlookr package.
It is problem with your data, not the code.

I found the cause of the problem and reproduced it with data and hints from [**Issues with density plots in R using geom_density_2d_filled**](https://stackoverflow.com/questions/70291634/issues-with-density-plots-in-r-using-geom-density-2d-filled).

According to the following URL, the problem is that the bandwidth estimate of your data for which you are plotting the density plot is 0.:
[**Error using geom_density_2d() in R : Computation failed in `stat_density2d()`: bandwidths must be strictly positive**](https://stackoverflow.com/questions/53075331/error-using-geom-density-2d-in-r-computation-failed-in-stat-density2d-b)
 
I have already solved the "EDA report fails if target is numerical or integer variable" issue you requested.
So, I will close this issue and create a new issue that handles exception handling in the plot.relate() function.
